src/cuda/memtransfer_wrapper.cu

```diff
@@ -434,12 +434,13 @@ void freegpumemory(cufinufft_plan_t<T> *d_plan)
   utils::WithCudaDevice device_swapper(d_plan->opts.gpu_device_id);
   // Fixes a crash whewre the plan itself is deleted before the stream
   const auto stream = d_plan->stream;
-
-  CUDA_FREE_AND_NULL(d_plan->fw, stream, d_plan->supports_pools);
-  CUDA_FREE_AND_NULL(d_plan->fwkerhalf1, stream, d_plan->supports_pools);
-  CUDA_FREE_AND_NULL(d_plan->fwkerhalf2, stream, d_plan->supports_pools);
-  CUDA_FREE_AND_NULL(d_plan->fwkerhalf3, stream, d_plan->supports_pools);
-
+  if(!d_plan->opts.gpu_spreadinterponly) {
+    CUDA_FREE_AND_NULL(d_plan->fw, stream, d_plan->supports_pools);
+    CUDA_FREE_AND_NULL(d_plan->fwkerhalf1, stream, d_plan->supports_pools);
```

Comment:
Confused why when opts.gpu_spreadinterponly=true you don't free up fwkerhalf1,2,3,   since there will have been allocated in the plan stage still?

Reply:
We dont allocate it if gpu_spreadinterponly=True, right? That was my understanding

Reply:
Yes, but CUDA_FREE_AND_NULL should be safe on nullptr, if null does nothing so this check might be redundant

Reply:
Thats weird, but tests tend to fail if I dont have this. Lemme double check